### PR TITLE
[flink] Avoid deprecated SetupableStreamOperator

### PR DESCRIPTION
**Pull request by yunfengzhou-hub**
### Purpose

Linked issue: #4442

This PR fixes deprecated usage on Flink's `SetupableStreamOperator`, and replaces the function with `StreamOperatorFactory`.

### Tests

Existing test cases are used to verify this change.

### API and Format

This change does not affect API or storage format.

### Documentation

This change does not affect introduce a new feature.

### CODE REVIEW

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/FlinkCdcMultiTableSink.java [hunk 1]
```diff
@@ -129,10 +129,9 @@ public DataStreamSink<?> sinkFrom(
                         .transform(
                                 GLOBAL_COMMITTER_NAME,
                                 typeInfo,
-                                new CommitterOperator<>(
+                                new CommitterOperatorFactory<>(
                                         true,
                                         false,
-                                        commitChaining,
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/FlinkCdcMultiTableSink.java:135]: remove commitChaining field

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/PrepareCommitOperator.java [hunk 2]
```diff
@@ -54,7 +53,6 @@ public abstract class PrepareCommitOperator<IN, OUT> extends AbstractStreamOpera
 
     public PrepareCommitOperator(Options options) {
         this.options = options;
-        setChainingStrategy(ChainingStrategy.ALWAYS);
```
[the remark this answers is not in the document]
Comment (yunfengzhou-hub) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/PrepareCommitOperator.java:57]: `AbstractStreamOperator#getChainingStrategy` is used in `SimpleOperatorFactory` when the operator is subclass of `SetupableStreamOperator`. When `PrepareCommitOperator` does not inherit from `SetupableStreamOperator`, `SimpleOperatorFactory` will use `ChainingStrategy.DEFAULT_CHAINING_STRATEGY = ALWAYS`.

[the remark this answers is not in the document]
Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/PrepareCommitOperator.java:57]: Can you add tests for this? To check chainable.

**Comment by JingsongLi** [comment]
Thanks @yunfengzhou-hub , +1

**Pull request merged**